aws/resource_aws_instance.go

```diff
@@ -818,7 +818,7 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error setting tags: %s", err)
 	}
 
-	volumeTags, err := readVolumeTags(conn, d.Id())
+	volumeTags, err := readRootVolumeTags(d, instance, conn)
```

Comment:
Main idea: work with root volume only, ignore tag changes for all other `ebs_block_device`s. 
It's stated [in the docs](https://github.com/terraform-providers/terraform-provider-aws/blame/master/website/docs/r/instance.html.markdown#L158):
> Currently, changes to the `ebs_block_device` configuration of _existing_ resources cannot be automatically detected by Terraform. To manage changes and attachments of an EBS block to an instance, use the `aws_ebs_volume` and `aws_volume_attachment` resources instead.

I believe, technically it's impossible to describe change for N volumes by single diff of `volume_tags`.

Reply:
Not a developer here, but maybe update 
https://www.terraform.io/docs/providers/aws/r/instance.html#volume_tags

`(Optional) A map of tags to assign to the devices created by the instance at launch time.`
to
`(Optional) A map of tags to assign to the root devices. For other ebs block devices, use the tag feature of the aws_ebs_volume resource`